code/modules/mob/living/carbon/human/say.dm

```diff
@@ -297,4 +297,24 @@
 				if(length(cword))
 					rearranged += cword
 			message ="[prefix][jointext(rearranged," ")]"
+		if(oxygen_alert) //Gasping is a mutation, right?

```

Comment:
Lohikar is correcto.

Check the variable `losebreath` instead. It's increased every time a person is unable to breathe. For reference, some chemicals use the condition of `if (losebreath >= 10)`, though you'll need to see where the nominal point is.